discovery/discovery.go

```diff
@@ -73,12 +79,12 @@ func parse(rawurl string) (string, string) {
 }
 
 // New is exported
-func New(rawurl string, heartbeat uint64) (Discovery, error) {
+func New(rawurl string, heartbeat uint64, tls *TLS) (Discovery, error) {
```

Comment:
Since currently none of ou discovery service supports TLS, can we not change the prototype of this func but change it when we will add TLS ?